Fixes TypeError check in Unmarshal

BrianBland opened the pull request:
decoder.terrors can be empty but non-nil if the unmarshaler function is called multiple times in UnmarshalYAML.
This fixes the case where d.terrors is empty but has a non-zero capacity, which previously resulted in returning a TypeError with an empty list of error strings.

niemeyer commented:
Thanks for submitting this. Can we have a test covering the need for the change, please?

Also, would you mind to sign the contribution agreement if you haven't already done so?

```
https://www.ubuntu.com/legal/contributors
```

It's straightforward/non-draconian, and easy to sign online.

Thanks!

BrianBland committed:
Fixes TypeError check in Unmarshal

decoder.terrors can be empty but non-nil if the unmarshaler function is called multiple times in UnmarshalYAML.
This fixes the case where d.terrors is empty but has a non-zero capacity, which previously resulted in returning a TypeError with an empty list of error strings.

BrianBland commented:
Ok, I've added a test which fails before the patch and signed the agreement.

BrianBland commented:
Can I get an update on this PR? Thanks.

stevvooe commented:
@niemeyer Ping!

niemeyer commented:
Thanks, and sorry for the long delay on this.

stevvooe commented:
@niemeyer Thank you!
